config/application_space_test.go

```diff
@@ -0,0 +1,321 @@
+// Copyright (c) 2023 IndyKite
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config_test
+
+import (
+	"context"
+	"errors"
+
+	"github.com/golang/mock/gomock"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/protobuf/types/known/wrapperspb"
+
+	"github.com/indykite/indykite-sdk-go/config"
+	sdkerrors "github.com/indykite/indykite-sdk-go/errors"
+	configpb "github.com/indykite/indykite-sdk-go/gen/indykite/config/v1beta1"
+	"github.com/indykite/indykite-sdk-go/test"
+	configmock "github.com/indykite/indykite-sdk-go/test/config/v1beta1"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("AppSpace", func() {
+	var (
+		ctx          context.Context
+		mockCtrl     *gomock.Controller
+		mockClient   *configmock.MockConfigManagementAPIClient
+		configClient *config.Client
+	)
+
+	BeforeEach(func() {
+		ctx = context.Background()
+		mockCtrl = gomock.NewController(GinkgoT())
+		mockClient = configmock.NewMockConfigManagementAPIClient(mockCtrl)
+
+		var err error
+		configClient, err = config.NewTestClient(ctx, mockClient)
+		Ω(err).To(Succeed())
+	})
+
+	Describe("AppSpace", func() {
+		DescribeTable("Request error during Read ",
+			func(req *configpb.ReadApplicationSpaceRequest, causeMatcher, msgMatcher, codeMatcher OmegaMatcher) {
+				resp, err := configClient.ReadApplicationSpace(ctx, req)
+				Expect(err).To(HaveOccurred())
+				Expect(resp).To(BeNil())
+
+				var clientErr *sdkerrors.ClientError
+				Expect(errors.As(err, &clientErr)).To(BeTrue(), "is client error")
+				Expect(clientErr.Unwrap()).To(causeMatcher)
+				Expect(clientErr.Message()).To(msgMatcher)
+				Expect(clientErr.Code()).To(codeMatcher)
+			},
+			Entry("nil request", nil, BeNil(), Equal("invalid nil request"), Equal(codes.InvalidArgument)),
+		)
+
+		Context("When sending a short request", func() {
```

Comment:
What is "short request"? And maybe you don't need that context, when wrapping 1 single `It()`